src/icastats_shared.c

```diff
@@ -53,7 +53,9 @@ int stats_mmap(int user)
 		sprintf(shm_id, "icastats_%d",
 			user == -1 ? geteuid() : (uid_t)user);
 
-		stats_shm_handle = shm_open(shm_id, O_CREAT | O_RDWR,
+		stats_shm_handle = shm_open(shm_id, O_RDWR,  S_IRUSR | S_IWUSR);
+		if (stats_shm_handle == NOT_INITIALIZED)
+			stats_shm_handle = shm_open(shm_id, O_CREAT | O_RDWR,
```

Comment:
If I execute this as root for another user, that user will later not be able to open its shared memory segment since it is created by root with only read and write permissions for user root...

Reply:
there is this block below, isn't this changing the owner to the user it created it for?

```
		if (user > 0 && geteuid() == 0) {
			if (fchown(stats_shm_handle, user, user) == -1) {
				close(stats_shm_handle);
				return -1;
			}
		}
```

Reply:
Yes, missed that in the diff...

Reply:
BTW: We theoretically have the same problem with OCK statistics. There I did have a fchown as well, but you commented that it would it be required :-) But actually, the code in OCK will never open it for another user, so this is only a theoretical problem there.